sympy/printing/tests/test_latex.py

```diff
@@ -2834,9 +2834,17 @@ def test_pickleable():
     import pickle
     assert pickle.loads(pickle.dumps(latex)) is latex
 
-def test_printing_latex_array_expressions():
-    assert latex(ArraySymbol("A", (2, 3, 4))) == "A"
-    assert latex(ArrayElement("A", (2, 1/(1-x), 0))) == "{{A}_{2, \\frac{1}{1 - x}, 0}}"
+
+def test_printing_ArraySymbol():
+    A = ArraySymbol("A", shape=(2, 3, 4))
+    assert latex(A) == "A"
+
+
+def test_printing_ArrayElement():
+    element = ArrayElement("A", indices=(2, 1 / (1 - x), 0))
```

Comment:
Why are non-symbols allows for indices?

Reply:
Allowing expressions that are always integers if the symbols are integers makes sense to me, e.g. `2*x+1`, but for the example in this test, I would expect an error to be thrown.

Reply:
Yeah `1/(1-x)` is a bit weird, I'll rewrite this test to something less surprising. The test case was already there before this PR btw.

I don't think there's much to check here though in order to raise an exception. What should the exception be in this case?

Reply:
> What should the exception be in this case?

Ah ok one could check for `is_Integer`. Does that make sense?

Reply:
It would make sense to check `is_integer` not to be `False` (note: lowercase _integer_, because capital _Integer_ matches the Python class). If `is_integer` returns `None`, it should still be allowed (it means that it could not be determined, but it can still be an integer).

Reply:
In this particular case, `None` would be returned correct? Since `1/(1-x)` would return an integer if `x=0`.

Reply:
> In this particular case, `None` would be returned correct?

Indeed.. Just played around a bit and this `1/(1-x)` would indeed have `is_integer is None`. So such a check wouldn't fish it out.

What would be a good test case from practice? It's better not to implement things if there are no ways to test it. And I can't come up with a good one other than simply feeding an index with assumptions set.

Reply:
Well, SymPy has the concept of _fuzzy booleans_, that is, booleans can have values `True`, `False`, and `None`. The last one, `None`, is meant for expressions whose truth could not be determined, either because it depends on some variables, or because the algorithm wasn't good enough to determine the truth of the statement.

Reply:
Ok so do b5da9d10e5 and 9286774c0a settle this?

Reply:
I think so.